src/exportImage.py

```diff
@@ -10,28 +11,46 @@
 from src.helper import Helper
 
 
-def get_zoom_resolution(zoom_to_resolution_dict, zoom_level):
-    if f'{zoom_level}' in zoom_to_resolution_dict:
-        resolution = zoom_to_resolution_dict[f'{zoom_level}']
-        return resolution
-    else:
-        raise Exception(f'No such zoom level. got: {zoom_level}')
-
-
 def is_valid_zoom_for_area(resolution, bbox):
     top_right_lat = bbox[3]
     top_right_lon = bbox[2]
     bottom_left_lat = bbox[1]
     bottom_left_lon = bbox[0]
 
     # Check if bbox width and height are at least at the resolution of a pixle at the wanted zoom level
-    return (top_right_lon - bottom_left_lon) >= float(resolution) and (top_right_lat - bottom_left_lat) >= float(resolution)
+    return (top_right_lon - bottom_left_lon) > float(resolution) and (top_right_lat - bottom_left_lat) > float(resolution)
+
+
+def is_valid_overview_factor(dataset, ov_factor):
+    """
+    Check if a given overview factor is valid for the dataset.
+    """
+    # Use gdal Dataset attributes (https://gdal.org/python/osgeo.gdal.Dataset-class.html)
+    return dataset.RasterXSize / ov_factor >= 8 and dataset.RasterYSize / ov_factor >= 8
+
+
+def calculate_overviews(dataframe, max_zoom):
+    """
+    Calculate overview factors for the given dataframe.
+
+    return an array of valid overview factors (2 ^ i) for the dataset
+    """
+    return [2 ** i for i in range(1, max_zoom + 1) if is_valid_overview_factor(dataframe, 2 ** i)]
+
+
+class GDALBuildOverviewsResponse(Enum):
+    CE_None = 0
+    CE_Debug = 1
+    CE_Warning = 2
+    CE_Failure = 3
+    CE_Fatal = 4
 
 
 class ExportImage:
     def __init__(self):
         self.log = Logger.get_logger_instance()
         self.__helper = Helper()
+        self.warp_percent = 70
```

Comment:
do you think to leave 1% for the index creation and actual s3 upload?

Reply:
Added